.github/workflows/main.yml

```diff
@@ -279,6 +284,36 @@ jobs:
         working-directory: examples/profiling
       - run: sccache --show-stats
 
+  test-crates:
+    if: needs.changes.outputs.test-crates == 'true'
+    needs: changes
+    runs-on: [self-hosted, prod, cpu, "${{ matrix.os }}"]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [Linux, macOS]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: 'risc0'
+      - uses: actions/checkout@v4
+        with:
+          repository: risc0/RustCrypto-crypto-bigint
+          path: 'RustCrypto-crypto-bigint'
+      - uses: ./risc0/.github/actions/rustup
+      - uses: ./risc0/.github/actions/sccache
+        with:
+          key: ${{ matrix.os }}-default
+      - name: Install risczero toolchain
+        run: |
+          cargo run --bin cargo-risczero --no-default-features -- risczero install --version $RISC0_TOOLCHAIN_VERSION
+          cargo install --force --path risc0/cargo-risczero -F experimental
+        working-directory: risc0
+      - name: run cargo risczero test
+        working-directory: ${{ github.workspace }}/RustCrypto-crypto-bigint
```

Comment:
Can/should we add this to an existing job rather than having to spin up a new machine? How long does it take to run this test? Do we need to run it on all targets or is Linux sufficient?

Basically we need to be mindful of the number of CI machines we have available and the impact this has on build times.

Reply:
@flaub sorry for the late response - I somehow missed this message. This test doesn't take a very long time and we can include it as a part of another workflow. I think linux is sufficient but doesn't hurt to run it on mac. We don't care about running this on metal/cuda.

How about we rename the "check-template" job to "test cargo-risczero commands" and have that workflow run tests involving cargo risczero?